tests/test_ntl.py

```diff
@@ -117,3 +118,146 @@ def test_fft_batch_interpolate(galois_field, galois_field_roots):
 
     # Then
     assert interp_polys == polys
+
+
+def test_evaluate(galois_field, polynomial):
+    # Given
+    p = galois_field.modulus
+    coeffs = [1, 2, 3, 4]
+    poly = polynomial(coeffs)
+    xs = [random.randint(0, p - 1) for _ in range(100)]
+
+    # When
+    ys = [evaluate(coeffs, x, p) for x in xs]
+
+    # Then
+    assert ys == [poly(x).value for x in xs]
+
+
+def test_gao_interpolate():
+    int_msg = [2, 3, 2, 8, 7, 5, 9, 5]
+    k = len(int_msg)  # length of message
+    n = 22  # size of encoded message
+    p = 53  # prime
+    t = k - 1  # degree of polynomial
+
+    x = list(range(n))
+    encoded = [sum(int_msg[j] * pow(x[i], j, p) for j in range(k)) % p
+               for i in range(n)]
+
+    # Check decoding with no errors
+    decoded, _ = gao_interpolate(x, encoded, k, p)
+    assert (decoded == int_msg)
+
+    # Corrupt with maximum number of erasures:
+    cmax = n - 2 * t - 1
```

Comment:
Aren't these better served as separate test cases? I would advice against checking multiple functionalities in one test since the first failure masks all others.

Reply:
Same comment for the test cases below.